File: 189/main.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {
  if (k > nums.length) {
    k = k % nums.length
  }
  let length = nums.length
  nums.splice(nums.length, 0, ...nums.slice(0, nums.length - k))
  nums.splice(0, length - k)
}

if (process.env.LZS) { // local test
  let arr = [1, 2, 3, 4, 5, 6, 7]
  rotate(arr, 3)
  console.log(arr)
}

File: 189/main.py
class Solution:
    def rotate(self, nums, k):
        if k > len(nums):
            k = k % len(nums)
        length = len(nums)
        nums.extend(nums[0:len(nums) - k])
        nums.reverse()
        for _ in range(length - k):
            nums.pop()
        nums.reverse()


if __name__ == '__main__':  # local test
    import os
    if os.getenv('LZS'):
        s = Solution()
        nums = [1, 2, 3, 4, 5, 6, 7]
        s.rotate(nums, 3)
        print(nums)
    # assert nums == [5, 6, 7, 1, 2, 3, 4]
